src/execution_plan/execution_plan_reduce_to_apply.c

```diff
@@ -7,29 +7,27 @@
 static OpBase *_buildMatchBranch(ExecutionPlan *plan, const cypher_astnode_t *path) {
 	AST *ast = QueryCtx_GetAST();
 
-	rax *bound_vars = NULL;
-	const char **arguments = NULL;
-	if(plan->root) {
-		bound_vars = raxNew();
-		// Rather than cloning the record map, collect the bound variables along with their
-		// parser-generated constant strings.
-		ExecutionPlan_BoundVariables(plan->root, bound_vars);
-		// Prepare the variables for populating the Argument ops we will build.
-		arguments = ExecutionPlan_BuildArgumentModifiesArray(bound_vars);
-	}
+	rax *bound_vars = raxNew();
+	// Rather than cloning the record map, collect the bound variables along with their
+	// parser-generated constant strings.
+	ExecutionPlan_BoundVariables(plan->root, bound_vars);
+	// Prepare the variables for populating the Argument ops we will build.
+	const char **arguments = ExecutionPlan_BuildArgumentModifiesArray(bound_vars);
 
 	// Initialize an ExecutionPlan that shares this plan's Record mapping.
 	ExecutionPlan *match_branch_plan = ExecutionPlan_NewEmptyExecutionPlan();
 	ExecutionPlan_AppendSubExecutionPlan(plan, match_branch_plan);
 
 	// We have bound variables, build an Argument op that represents them.
 	OpBase *argument = NewArgumentOp(match_branch_plan, arguments);
+	array_free(arguments);
 	match_branch_plan->root = argument;
```

Comment:
Consider exposing `_ExecutionPlan_UpdateRoot` defiantly not a must

Reply:
I leave it for now